packages/react-native-renderer/src/ReactNativeComponent.js

```diff
@@ -37,36 +36,48 @@ export default function(
    * components that are not directly backed by a native view. For more
    * information, see [Direct Manipulation](docs/direct-manipulation.html).
    *
+   * Subclasses must set the `ref` on the root element to `this.nativeRef`.
+   *
    * @abstract
    */
   class ReactNativeComponent<Props, State = void> extends React.Component<
     Props,
     State,
   > {
+    _nativeTag: ?number;
+
     /**
      * Due to bugs in Flow's handling of React.createClass, some fields already
      * declared in the base class need to be redeclared below.
      */
     props: Props;
     state: State;
 
+    /**
+     * Subclasses should set the `ref` prop on the root element to this.
+     */
+    nativeRef = (nativeElementRef: ?any): void => {
+      this._nativeTag =
+        nativeElementRef != null ? nativeElementRef._nativeTag : null;
```

Comment:
`_nativeTag` is a private implementation detail of the stateNode in this particular renderer. I want to change it in Fabric. I also wanted it to be lazy and this forces it to be eager. You're better off storing the actual ref and then call `findNodeHandle` on it.